features/storage/blockdevice/BufferedBlockDevice.cpp

```diff
@@ -110,36 +124,52 @@ int BufferedBlockDevice::sync()
 
 int BufferedBlockDevice::read(void *b, bd_addr_t addr, bd_size_t size)
 {
-    MBED_ASSERT(_cache);
+    MBED_ASSERT(_write_cache);
```

Comment:
Should this be?
```suggestion
    MBED_ASSERT(_read_cache);
```

Also should there be a similar assert in BlockDevice::program?

Reply:
@davidsaada ^^^

Reply:
@cmonr, given the dependencies, it's probably better to just move forward with testing. The assert is minor and can be fixed after the release (or if @davidsaada has to make further changes).

Reply:
@geky Alright, that sounds reasonable.

Reply:
Used those damned netsocket test failures to fix these asserts.